cvmfs/bundle.cc

```diff
@@ -0,0 +1,87 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#include "bundle.h"
+
+#include <string>
+
+#include "json_document.h"
+#include "pack.h"
+#include "util/exception.h"
+#include "util/posix.h"
+#include "util/string.h"
+
+ObjectPack* Bundle::CreateBundle(const JSON* json_obj)
+{
+  const JSON* value = json_obj;
+  if (value->type != JSON_OBJECT) {
+    PANIC(kLogStderr, "JSON object not found");
+  } else {
+    // create an ObjectPack
+    ObjectPack* op = new ObjectPack();
+
+    if (op == NULL) {
+      PANIC(kLogStderr, "Insufficient memory");
+    }
+
+    value = (value->first_child);  // bundleid
+    value = (value->next_sibling);  // JSON array
+    if (value->first_child) {
+      value = value->first_child;
+      std::string filepath = std::string(value->string_value);
+
+      // if file doesn't exist then bundle is not created
+      if (!FileExists(filepath)) {
+        PrintWarning("File not found: " + filepath);
```

Comment:
We should decide here: either warn and skip (instead of returning) or we `PANIC`. I think we want to warn and skip.